Source/WebCore/rendering/RenderBlockFlow.cpp

```diff
@@ -4663,5 +4755,17 @@ bool RenderBlockFlow::tryComputePreferredWidthsUsingModernPath(LayoutUnit& minLo
     return true;
 }
 
+LayoutUnit RenderBlockFlow::blockFormattingContextInFlowContentHeight() const
+{
+    ASSERT(createsBlockFormattingContext());
+    // For block layout we should just be able to check the height of the last in flow box
+    auto lastChild = lastInFlowChildBox();
+    if (!lastChild)
+        return 0_lu;
+    auto lastChildMarginBoxLogicalHeight = isHorizontalWritingMode() ? lastChild->marginBoxRect().height() : lastChild->marginBoxRect().width();
+    // Child's margin box starting location + border box height + margin after size
+    return lastChild->marginBoxRect().y() + lastChildMarginBoxLogicalHeight;
```

Comment:
is marginBoxRect().y() is what you expect to be the correct value in vertical mode?

Reply:
Nope. I created `logicalMarginBoxTopForChild`and `logicalMarginBoxHeightForChild` to help with this so should return the correct values according to containing block's writing mode